### PR TITLE
refactor(cli): Remove cli/node dependency on cli/compat

### PR DESCRIPTION
**Pull request by bartlomieju**
In preparation for removal of `--compat` mode.

### CODE REVIEW

File: cli/node/mod.rs [hunk 1]
```diff
@@ -27,17 +27,94 @@ use once_cell::sync::Lazy;
...
+  "util/types",
+  "v8",
+  "vm",
+  "worker_threads",
+  "zlib",
+];
+
+// WARNING: Ensure this is the only deno_std version reference as this
+// is automatically updated by the version bump workflow.
+pub(crate) static STD_URL_STR: &str = "https://deno.land/std@0.153.0/";
```
Comment (bartlomieju) [inline cli/node/mod.rs:89]: I think I will move this to `cli/deno_std.rs`

File: tools/release/02_create_pr.ts [hunk 2]
```diff
@@ -36,7 +36,7 @@ function getPrBody() {
     `Please ensure:\n` +
     `- [ ] Target branch is correct\n` +
     `- [ ] Crate versions are bumped correctly\n` +
-    `- [ ] deno_std version is incremented in the code (see \`cli/compat/mod.rs\`)\n` +
+    `- [ ] deno_std version is incremented in the code (see \`cli/deno_std/mod.rs\`)\n` +
```
Comment (lucacasonato) [inline tools/release/02_create_pr.ts:39]:
```suggestion
    `- [ ] deno_std version is incremented in the code (see \`cli/deno_std.rs\`)\n` +
```

File: tools/release/release_doc_template.md [hunk 3]
```diff
@@ -100,7 +100,7 @@ verify on GitHub that everything looks correct.
   1. Checkout the branch the release is being made on.
   2. Manually run `./tools/release/01_bump_crate_versions.ts`
      1. Ensure the crate versions were bumped correctly
-     2. Ensure deno_std version was updated correctly in `cli/compat/mod.rs`
+     2. Ensure deno_std version was updated correctly in `cli/deno_std/mod.rs`
```
Comment (lucacasonato) [inline tools/release/release_doc_template.md:103]:
```suggestion
     2. Ensure deno_std version was updated correctly in `cli/deno_std.rs`
```

File: tools/release/02_create_pr.ts (hunk 2 above)
Comment (bartlomieju) [reply tools/release/02_create_pr.ts]: Thanks, fixed these typos

**Review by dsherret — APPROVED** [review]
LGTM. Nice change

**Pull request merged**